planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/src/scene_module/intersection/scene_intersection.cpp

```diff
@@ -184,27 +184,33 @@ bool IntersectionModule::modifyPathVelocity(
 }
 
 void IntersectionModule::cutPredictPathWithDuration(
-  autoware_perception_msgs::msg::DynamicObjectArray * objects_ptr, const double time_thr) const
+  autoware_auto_perception_msgs::msg::PredictedObjects * objects_ptr, const double time_thr) const
 {
   const rclcpp::Time current_time = clock_->now();
-  for (auto & object : objects_ptr->objects) {                    // each objects
-    for (auto & predicted_path : object.state.predicted_paths) {  // each predicted paths
-      std::vector<geometry_msgs::msg::PoseWithCovarianceStamped> vp;
-      for (auto & predicted_pose : predicted_path.path) {  // each path points
-        if ((rclcpp::Time(predicted_pose.header.stamp) - current_time).seconds() < time_thr) {
-          vp.push_back(predicted_pose);
+  for (auto & object : objects_ptr->objects) {                         // each objects
+    for (auto & predicted_path : object.kinematics.predicted_paths) {  // each predicted paths
+      // TODO(murooka) check this logic
```

Comment:
remove this TODO(murooka) if the upper other cutPredictPathWithDuration is correct